File: api/src/main/java/cz/jeme/bestium/api/util/KeyUtils.java
package cz.jeme.bestium.api.util;

import net.kyori.adventure.key.Key;
import org.jspecify.annotations.NullMarked;

@NullMarked
public final class KeyUtils {
    private KeyUtils() {
        throw new AssertionError();
    }

    public static String keyToModelName(final Key key) {
        return "bestium." + key.namespace() + "." + key.value();
    }
}

File: core/src/main/kotlin/cz/jeme/bestium/util/KeyUtils.kt
package cz.jeme.bestium.util

import net.kyori.adventure.key.Key
import net.minecraft.resources.Identifier
import org.bukkit.NamespacedKey

fun Key.toNamespacedKey(): NamespacedKey = this as? NamespacedKey ?: NamespacedKey.fromString(asString())!!

fun Key.toIdentifier(): Identifier = Identifier.fromNamespaceAndPath(namespace(), value())
